framework/PostProcessors/Metric.py

```diff
@@ -230,7 +263,63 @@ def collectOutput(self,finishedJob, output):
     if isinstance(evaluation, Runners.Error):
       self.raiseAnError(RuntimeError, "Job ", finishedJob.identifier, "failed!")
     outputDict = evaluation[1]
-    output.addRealization(outputDict)
+    # FIXME store the data in DataObjects
+    if isinstance(output, Files.File):
+      availExtens = ['xml']
+      outputExtension = output.getExt().lower()
+      if outputExtension not in availExtens:
+        self.raiseAMessage('Metric postprocessor did not recognize extension ".', str(outputExtension), '". The output will be dumped to a text file')
+      output.setPath(self._workingDir)
+      self.raiseADebug('Write Metric prostprocessor output in file with name: ', output.getAbsFile())
+      output.open('w')
```

Comment:
use ```with``` construction.
You do not close the file here

Reply:
removed, not need anymore